harness/determined/keras/callbacks.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict
+
+import tensorflow as tf
+
+
+class Callback(tf.keras.callbacks.Callback):  # type: ignore
+    """
+    A Determined implementation of Callback which supports additional new callbacks.
+    """
+
+    def on_validation_period_begin(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_validation_period_end(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_train_step_begin(self, total_batches_trained: int, logs: Dict) -> None:
+        pass
+
+    def on_train_step_end(self, total_batches_trained: int, logs: Dict) -> None:
+        pass
+
+    def get_state(self) -> Any:
+        return None
+
+    def load_state(self, state: Any) -> None:
+        pass
```

Comment:
non-blocking: we had also talked about adding `on_checkpoint_end()` callback similar to the estimators. Is the plan to add that in a separate PR?

Reply:
IMO this *is* that callback.

Reply:
Isn't this for loading state of the checkpoint while the `on_checkpoint_end()` is for users insert additional metadata or other files into checkpoints as we do [here](https://github.com/determined-ai/determined/blob/c97dc523fe14e452453617d363253dd2576738dd/harness/determined/estimator/_callback.py#L22)

Reply:
I guess my question is why do we expose both a `state_dict` and an `on_checkpoint_end`?

Is the answer is "to let people store arbitrary data as files?"  (that would make sense)

Reply:
yeah exactly, we expose `state_dict` to save the state of the callback and `on_checkpoint_end` to allow adding arbitrary files to checkpoints.